R/middleware.R

```diff
@@ -299,18 +299,37 @@ HandlerManager <- R6Class("HandlerManager",
 
           if (reqSize > maxSize) {
             return(list(status = 413L,
-              headers = list(
-                'Content-Type' = 'text/plain'
-              ),
+              headers = list('Content-Type' = 'text/plain'),
               body = 'Maximum upload size exceeded'))
           }
           else {
             return(NULL)
           }
         },
         call = .httpServer(
-          function (req) {
-            withLogErrors(handlers$invoke(req))
+          function(req,
+                   full = getOption("shiny.fullstacktrace", FALSE),
+                   offset = getOption("shiny.stacktraceoffset", TRUE)) {
+
+             withCallingHandlers(captureStackTraces(handlers$invoke(req)),
+               error = function(cond) {
+                 # Don't print shiny.silent.error (i.e. validation errors)
+                 if (inherits(cond, "shiny.silent.error")) return()
+                 if (isTRUE(getOption("show.error.messages"))) {
```

Comment:
I inherited this check from the `shiny.R` code (before I added to it): `if (isTRUE(getOption("show.error.messages"))) printError(cond)`. `show.error.messages` is a general R error handling option: from my understanding, if it is `FALSE`, no errors (strictly specking, nothing produced by `stop` is printed). I don't think it's really important whether we keep this or not because I don't think this is a widely used option. It doesn't hurt though, and keeps things consistent.